tests/unit/query_graph/on_demand_function/test_unary.py

```diff
@@ -48,7 +48,7 @@
         (IsNullNode(**NODE_PARAMS), "feat.isnull()", "pd.isna(feat)"),
         (
             CastNode(**NODE_PARAMS, parameters={"type": "str", "from_dtype": DBVarType.INT}),
-            "feat.map(lambda x: str(x) if pd.notnull(x) else x)",
+            "feat.map(lambda x: str(x) if pd.notnull(x) else x).astype(object)",
```

Comment:
Using `astype(object)` is to handle the case of `np.nan`, which won't work for `astype(str)`, is that correct?

Reply:
Yes, you're right. `s.astype(str)` will convert everything to `str`, which is not the behaviour we want here.